dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/LuceneQueryDateTimeFormatter.java

```diff
@@ -0,0 +1,253 @@
+
+                    for (int pos = 0; pos < tempStructureFields.size(); ) {
+
+                        if (tempStructureFields.get(pos).getFieldType()
+                                .equals(Field.FieldType.DATE_TIME.toString()) ||
+                                tempStructureFields.get(pos).getFieldType()
+                                        .equals(Field.FieldType.DATE.toString()) ||
+                                tempStructureFields.get(pos).getFieldType()
+                                        .equals(Field.FieldType.TIME.toString())) {
+                            ++pos;
+                        } else {
+                            tempStructureFields.remove(pos);
+                        }
+
+                    }
+
+                    dateFields.addAll(tempStructureFields);
+                }
+            }
+        }
+
+        String replace;
+        List<RegExMatch> matches;
+        String structureVarName;
+        for (String clause: clauses) {
+            for (Field field: dateFields) {
+
+                structureVarName = CacheLocator.getContentTypeCache()
+                        .getStructureByInode(field.getStructureInode()).getVelocityVarName()
+                        .toLowerCase();
+
+                if (clause.startsWith(structureVarName + "." + field.getVelocityVarName().toLowerCase() + ":") || clause.startsWith("moddate:")) {
+                    replace = new String(clause);
+                    if (field.getFieldType().equals(Field.FieldType.DATE_TIME.toString()) || clause.startsWith("moddate:")) {
+                        matches = RegEX.find(replace, "\\[(\\d{1,2}/\\d{1,2}/\\d{4}) TO ");
+                        for (RegExMatch regExMatch : matches) {
+                            replace = replace.replace("[" + regExMatch.getGroups().get(0).getMatch() + " TO ", "["
+                                    + regExMatch.getGroups().get(0).getMatch() + " 00:00:00 TO ");
+                        }
+
+                        matches = RegEX.find(replace, " TO (\\d{1,2}/\\d{1,2}/\\d{4})\\]");
+                        for (RegExMatch regExMatch : matches) {
+                            replace = replace.replace(" TO " + regExMatch.getGroups().get(0).getMatch() + "]", " TO "
+                                    + regExMatch.getGroups().get(0).getMatch() + " 23:59:59]");
+                        }
+                    }
+
+                    //if structureVarName or field.getVelocityVarName() has numbers, we need to split the clause to format only dates
+                    if (clause.startsWith(
+                            structureVarName + "." + field.getVelocityVarName().toLowerCase()
+                                    + ":")) {
+                        replace = structureVarName + "." + field.getVelocityVarName()
+                                .toLowerCase() + ":" + replaceDateTimeFormatInClause(
+                                replace.substring(replace.indexOf(":") + 1));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [String.indexOf(char) is faster than String.indexOf(String).](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4545506)